apps/backoffice-v2/src/lib/blocks/hooks/useDirectorsBlocks/useDirectorsBlocks.tsx

```diff
@@ -148,33 +244,45 @@ export const useDirectorsBlocks = ({
                 })
                 .build()
                 .flat(1);
+
+              return approvedBadgeBlock;
+            } else {
+              if (decisionStatus !== 'revision') {
+                const approveButtonBlock = createBlocksTyped()
+                  .addBlock()
+                  .addCell({
+                    type: 'callToActionLegacy',
+                    value: {
+                      text: 'Approve',
+                      props: {
+                        decision: 'approve',
+                        id: document.id,
+                        contextUpdateMethod: 'director',
+                        disabled:
+                          (!isDoneWithRevision && Boolean(document?.decision?.status)) || noAction,
+                        workflow,
+                        onReuploadNeeded,
+                      },
+                    },
+                  })
+                  .build()
+                  .flat(1);
+
+                return approveButtonBlock;
+              }
             }
 
-            return createBlocksTyped()
-              .addBlock()
-              .addCell({
-                type: 'callToAction',
-                value: {
-                  text: 'Approve',
-                  onClick: onMutateApproveTaskById({
-                    taskId: document.id,
-                    contextUpdateMethod: 'director',
-                  }),
-                  props: {
-                    disabled:
-                      (!isDoneWithRevision && Boolean(document?.decision?.status)) ||
-                      noAction ||
-                      isLoadingApproveTaskById ||
-                      !caseState.actionButtonsEnabled,
-                    size: 'wide',
-                    variant: 'success',
-                  },
-                },
-              })
-              .build()
-              .flat(1);
+            return undefined;
           };
 
+          const documentHeading = [
+            getReuploadStatusOrAction(document?.decision?.status, workflow),
+            getReUploadedNeededAction(document?.decision?.status, workflow),
+            getDecisionStatusOrAction(document?.decision?.status, workflow),
+          ]
+            .filter(Boolean)
+            .map(block => block?.flat(1)[0]);
```

Comment:
Can you explain this code?

Reply:
Functions are returning `undefined` in some cases. So they are should be filtered out. You can see example at 1st and 2nd screenshots.

Reply:
Let's revisit the control flow of state to block.